core/include/array/array_sorted_read_state.h

```diff
@@ -201,17 +202,14 @@ class ArraySortedReadState {
   /** AIO counter. */
   int aio_cnt_;
 
-  /** The AIO mutex conditions (one for each buffer). */
-  pthread_cond_t aio_cond_[2];
+  /** Condition variables used in AIO. */
+  std::condition_variable aio_cv_[2];
```

Comment:
If there are only two possible values here (and the index implicitly represents some resource) why not just have two separate named variables for each AIO condition?   Will the number grow > 2 in the future?